### PR TITLE
Stream/v1 employee roles

### PR DESCRIPTION
**Pull request by jacobrobertbaca**
# Description of change
added roles stream
# Manual QA steps
 - added to test, ran tests
 
# Risks
 - low
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/test_pagination.py [hunk 1]
```diff
@@ -37,7 +37,8 @@ def testable_streams(self):
             {  # STREAMS NOT CURRENTY TESTABLE
                 'employees', # Requires production environment to create records
                 'modifier_lists',
-
+                'inventories',
```
Comment (asaf-erlich) [inline tests/test_pagination.py]: Why was this added to not-testable?

File: tap_square/client.py [hunk 2]
```diff
@@ -295,3 +302,46 @@ def get_payments(self, start_time, bookmarked_cursor):
...
             yield (result.body.get('payments', []), result.body.get('cursor'))
+
+
+    def get_batch_token(self, link): #pylint: disable=no-self-use
+        if link:
+            url = link[link.find('<')+1:link.find('>')]
+            parsed = urlparse.urlparse(url)
+            batch_token = parse_qs(parsed.query)['batch_token'][0]
+            return int(batch_token)
+        return None
```
Comment (asaf-erlich) [inline tap_square/client.py:313]: I'm confused why this returns none and why the while loop works the way it does. What's the flow here?

Comment (asaf-erlich) [reply tap_square/client.py:313]: Is it supposed to be the way it paginates?

File: tap_square/client.py [hunk 3]
```diff
@@ -295,3 +302,46 @@ def get_payments(self, start_time, bookmarked_cursor):
...
+
+        yield (result.json(), batch_token)
+
+        while batch_token:
+            params['batch_token'] = batch_token
+            with singer.http_request_timer('GET payments'):
+                result = requests.get(url, headers=headers, params=params)
+
+            # if result.is_error():
+            #     raise Exception(result.errors)
```
Comment (asaf-erlich) [inline tap_square/client.py]: Please uncomment this. All requests should check for errors and throw exception. Otherwise it hides failures.

File: tests/test_pagination.py (hunk 1 above)
Comment (jacobrobertbaca) [reply tests/test_pagination.py]: oh maybe it wasn't supposed to be there, it must have been a merge mistake.

File: tap_square/client.py (hunk 2 above)
Comment (jacobrobertbaca) [reply tap_square/client.py:313]: if there is no 'Link' in the header it means that the records are finished being returned, so i want batch_token to be None, because then it will break out of the while loop.

Comment (asaf-erlich) [reply tap_square/client.py:313]: This actually makes sense now that i think about it. If we were using type hints this method would return an `Optional[int]` (Optional in python just means it's either an int or a None).

Comment (asaf-erlich) [inline tap_square/client.py:313]: Can we change the order here to do if not link return None first? Like validate the input. Then do the url and batch token generation in the non-branch code?

**Review by asaf-erlich — APPROVED** [review]
lgtm with minor comment

File: tap_square/client.py (hunk 2 above)
Comment (luandy64) [reply tap_square/client.py:313]: Can we not use `find`? `requests` has utils functions for stuff like this
